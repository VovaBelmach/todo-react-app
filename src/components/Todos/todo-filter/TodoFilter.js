import React from "react";
import PropTypes from "prop-types";
import Button from "../../UI/Button/Button";
import styles from "./TodoFilter.module.css";
import {
  TODO_FILTER_ALL_BUTTON_NAME,
  TODO_FILTER_ACTIVE_BUTTON_NAME,
  TODO_FILTER_COMPLITED_BUTTON_NAME,
} from "../../../constants";

const TodoFilter = (props) => {
  const onClickHandler = (event) => {
    props.onFilterHandler(event.target.value);
  };

  return (
    <>
      <ul className={styles["todo-filter"]}>
        <li>
          <Button onClick={onClickHandler} value={TODO_FILTER_ALL_BUTTON_NAME}>
            {TODO_FILTER_ALL_BUTTON_NAME}
          </Button>
        </li>
        <li>
          <Button
            onClick={onClickHandler}
            value={TODO_FILTER_ACTIVE_BUTTON_NAME}
          >
            {TODO_FILTER_ACTIVE_BUTTON_NAME}
          </Button>
        </li>
        <li>
          <Button
            onClick={onClickHandler}
            value={TODO_FILTER_COMPLITED_BUTTON_NAME}
          >
            {TODO_FILTER_COMPLITED_BUTTON_NAME}
          </Button>
        </li>
      </ul>
    </>
  );
};

TodoFilter.propTypes = {
  onFilterHandler: PropTypes.func,
};

export default TodoFilter;
