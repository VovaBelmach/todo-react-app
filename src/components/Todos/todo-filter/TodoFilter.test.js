import { render, screen, fireEvent } from "@testing-library/react";
import TodoFilter from "./TodoFilter";
import {
    TODO_FILTER_ALL_BUTTON_NAME,
    TODO_FILTER_ACTIVE_BUTTON_NAME,
    TODO_FILTER_COMPLITED_BUTTON_NAME,
  } from "../../../constants";

describe("TodoFilter", () => {
  test("renders all filter button", () => {
    // Arrange
    render(<TodoFilter />);

    // Assert
    const allButton = screen.getByText(TODO_FILTER_ALL_BUTTON_NAME);
    expect(allButton).toBeInTheDocument();
  });

  test("renders active filter button", () => {
    // Arrange
    render(<TodoFilter />);

    // Assert
    const activeButton = screen.getByText(TODO_FILTER_ACTIVE_BUTTON_NAME);
    expect(activeButton).toBeInTheDocument();
  });

  test("renders completed filter button", () => {
    // Arrange
    render(<TodoFilter />);

    // Assert
    const completedButton = screen.getByText(TODO_FILTER_COMPLITED_BUTTON_NAME);
    expect(completedButton).toBeInTheDocument();
  });

  test("calls onFilterHandler with correct value when All button is clicked", () => {
    // Arrange
    const onFilterHandler = jest.fn();
    render(<TodoFilter onFilterHandler={onFilterHandler} />);
    
    // Act
    const allButton = screen.getByText(TODO_FILTER_ALL_BUTTON_NAME);
    fireEvent.click(allButton);

    // Assert
    expect(onFilterHandler).toHaveBeenCalledWith(TODO_FILTER_ALL_BUTTON_NAME);
  });

  test("calls onFilterHandler with correct value when Active button is clicked", () => {
    // Arrange
    const onFilterHandler = jest.fn();
    render(<TodoFilter onFilterHandler={onFilterHandler} />);
    
    // Act
    const activeButton = screen.getByText(TODO_FILTER_ACTIVE_BUTTON_NAME);
    fireEvent.click(activeButton);

    // Assert
    expect(onFilterHandler).toHaveBeenCalledWith(TODO_FILTER_ACTIVE_BUTTON_NAME);
  });

  test("calls onFilterHandler with correct value when Completed button is clicked", () => {
    // Arrange
    const onFilterHandler = jest.fn();
    render(<TodoFilter onFilterHandler={onFilterHandler} />);
    // Act
    const completedButton = screen.getByText(TODO_FILTER_COMPLITED_BUTTON_NAME);
    fireEvent.click(completedButton);

    // Assert
    expect(onFilterHandler).toHaveBeenCalledWith(TODO_FILTER_COMPLITED_BUTTON_NAME);
  });
});
