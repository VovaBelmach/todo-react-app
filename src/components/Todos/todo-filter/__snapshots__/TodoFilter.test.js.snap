// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFilter should renders correctly 1`] = `
<ul
  className="todo-filter"
>
  <li>
    <button
      className="button "
      onClick={[Function]}
      type="button"
      value="All"
    >
      All
    </button>
  </li>
  <li>
    <button
      className="button "
      onClick={[Function]}
      type="button"
      value="Active"
    >
      Active
    </button>
  </li>
  <li>
    <button
      className="button "
      onClick={[Function]}
      type="button"
      value="Completed"
    >
      Completed
    </button>
  </li>
</ul>
`;
